static/js/app.js

```diff
@@ -264,37 +264,46 @@ $(function () {
                 }
             },this);
 
+            //goes over all overlapping markers
             _.each(this.oms.markersNearAnyOtherMarker(), function (marker) {
-                if(marker.view.model.get("currentlySpiderfied")){
-                    marker.setTitle(marker.view.getTitle("single"));
-                }else{
-                    marker.setTitle(marker.view.getTitle("multiple"));
-                }
                 var firstMember = marker.view.model;
-                if (!firstMember.get("groupID")) {
-                    firstMember.set("groupID", groupID);
-                    var firstMemberSeverity = firstMember.get('severity');
+                var firstMemberGroupId = firstMember.get("groupID");
+                if (!firstMemberGroupId) {
+                    firstMemberGroupId = groupID;
+                    firstMember.set("groupID", firstMemberGroupId);
+                    var groupSeverity = firstMember.get('severity');
                     var firstMemberOpacity = firstMember.get("locationAccuracy") == 1 ? 'opaque' : 1;
-                    groupsData.push({severity: firstMemberSeverity, opacity: firstMemberOpacity});
+                    groupsData.push({severity: groupSeverity, opacity: firstMemberOpacity, quantity: 1});
 
+                    //goes over all markers which overlapping 'firstMember', and get the 'max' severity
                     _.each(this.oms.markersNearMarker(marker), function (markerNear) {
                         var markerNearModel = markerNear.view.model;
-                        markerNearModel.set("groupID", groupID);
-                        if ((firstMemberSeverity != markerNearModel.get('severity'))) {
-                            groupsData[groupsData.length - 1].severity = SEVERITY_VARIOUS;
+                        markerNearModel.set("groupID", firstMemberGroupId);
+                        var currentMarkerNearSeverity = markerNearModel.get('severity');
+                        // severity is an enum, when it's lower then it's more severe
+                        if (currentMarkerNearSeverity < groupSeverity) {
+                            groupsData[firstMemberGroupId - 1].severity = currentMarkerNearSeverity;
```

Comment:
Since this `groupsData[firstMemberGroupId - 1]` object is referenced so many times, why not extract it to a variable?